### ISSUE TITLE
resolve symlinks at build time

### ISSUE THREAD

**Issue by edrex** [sintaxi/harp#255]
**TL;DR** I would like to see symbolic links to files which are either hidden via a leading `_` or outside the source directory resolved at `compile` time, to have parity with `server`.

My main use case (there are likely others):
- I have a bower `components` directory, but I don't want everything inside to be in the output. I would like to keep components outside the source tree or in an `_`-omitted dir and then symlink only the parts I need

Symlink cases:
1. link to file which is present in the output
2. link to file which is outside the source directory
3. link to file which is not in the output because it is omitted via a leading `_`
4. link to file which is not in the output because:
   - it is a source file whose extension changes on compile
   - it doesn't exist in the source or dest
- 1 works currently: the symlink is preserved.
- 2 and 3 work in `server` mode but not in compile.
- 4 is the user's fault and probably the correct thing is to preserve the dangling symlink.

**Comment by clayadavis** [comment]
This is an issue for me too :+1:
